controllers/mysql_container.go

```diff
@@ -457,3 +459,23 @@ func updateContainerWithOverwriteContainers(cluster *mocov1beta2.MySQLCluster, c
 		}
 	}
 }
+
+// updateContainerWithTimezoneDataResources overrides memory resources when InitializeTimezoneData is enabled
```

Comment:
I'd prefer to increase the default moco-init memory to 512Mi rather than adding the updateContainerWithTimezoneDataResources function. This eliminates conditional complexity and provides sufficient memory for all use cases.